url9d04094941_petermikitsh_PizzaDeliverySystem_tgz-pJ8-testdrivers_ManagerReportTestDriverJ8/src/neworder/addpizza/AddPizzaViewGUI.java

```diff
@@ -0,0 +1,889 @@
+		_quantityPanel.add(quantityLabel, quantityLabelConstraints);
+		
+		GridBagConstraints quantityFieldConstraints = new GridBagConstraints();
+		quantityFieldConstraints.insets = new Insets(0,0,0,100);
+		_quantityPanel.add(_quantityField, quantityFieldConstraints);
+		
+		GridBagConstraints addOrderButtonConstraints = new GridBagConstraints();
+		//addOrderButtonConstraints.ipadx = 10;
+		_quantityPanel.add(_addToOrderButton, addOrderButtonConstraints);
+		
+		validQuantity = true;
+	}
+
+	
+	/**
+	 * Called when a button is pressed
+	 * 
+	 * @param aEvent
+	 */
+	public void actionPerformed(ActionEvent aEvent) {
+		
+		// Check if the event was caused by a button.
+		if( aEvent.getSource() instanceof JButton ) {
+
+			JButton sourceButton = (JButton)aEvent.getSource();
+			
+			if( sourceButton.getText().equals( 
+					AddPizzaViewGUI.SMALL_BUTTON ) ) {
+				size = PizzaFoodItem.PizzaSize.PizzaSizeSmall;
+				_selectedSizeButton.setBackground(null);
+				smallPizzaButton.setBackground(Color.lightGray);
+				_selectedSizeButton = smallPizzaButton;
+			}
+			else if( sourceButton.getText().equals( 
+					AddPizzaViewGUI.MEDIUM_BUTTON ) ) {
+				size = PizzaFoodItem.PizzaSize.PizzaSizeMedium;
+				_selectedSizeButton.setBackground(null);
+				mediumPizzaButton.setBackground(Color.lightGray);
+				_selectedSizeButton = mediumPizzaButton;
+			}
+			else if( sourceButton.getText().equals( 
+					AddPizzaViewGUI.LARGE_BUTTON ) ) {
+				size = PizzaFoodItem.PizzaSize.PizzaSizeLarge;
+				_selectedSizeButton.setBackground(null);
+				largePizzaButton.setBackground(Color.lightGray);
+				_selectedSizeButton = largePizzaButton;
+			}
+			else if( sourceButton.getText().equals( 
+					AddPizzaViewGUI.ADD_MORE_TOPPINGS_BUTTON ) ) {
+				ToppingTableCell toppingOption = new ToppingTableCell(this);
+				toppingsOptionsCells.add( toppingOption );
+				_toppingsOptionsTable.updateTable();
+				_toppingsOptionsTable.scrollToBottom();
+				updateProgressBars();
+				
+				
+			}
+			else if( sourceButton.getText().equals( 
+					AddPizzaViewGUI.ADD_TO_ORDER_BUTTON_NAME) ||
+					sourceButton.getText().equalsIgnoreCase(
+							"Done")) {
+					sendToController();
+				
+			}
+			else if( sourceButton.getText().equals( 
+					PDSViewManager.BACK_BUTTON_NAME ) ) {
+				
+				if(modifying){
+					sendToController(originalPizzaToModify);
+				}
+				else{
+					controller.respondToInput( AddPizzaView.BACK_KEY ,
+						AddPizzaInChan.ICGoBack );
+				}
+
+			}
+		}
+			
+		
+		
+	}
+	
+	/**
+	 * Function that handles processing the order by sending all of the
+	 * relevent information to the controller.
+	 */
+	public void sendToController(){
+		
+		String sizeKey = "";
+		
+		switch(size){
+		case PizzaSizeSmall:
+			sizeKey = AddPizzaView.SMALL_KEY;
+			break;
+			
+		case PizzaSizeMedium:
+			sizeKey = AddPizzaView.MEDIUM_KEY;
+			break;
+			
+		case PizzaSizeLarge:
+			sizeKey = AddPizzaView.LARGE_KEY;
+			break;
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 577:589, 617:629.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.